### PR TITLE
Permission Set List API

### LINKED ISSUE #69: permission_set list api
**Is your feature request related to a problem? Please describe.**
As a admin of the organisation i would like to view the list of permission sets available in the org.

image

**Describe the solution you'd like**
1. Create an api: `[get] /api/org/(org-id)/permission_set/` which lists the permissions sets available under that org.
1. It will return an array of permission_set objects which contain 
    1. permission_set.id 
    1. permission_set.name
(no need of role and actual permission info here as we can see from img)
1. only **authenticated users** can access it.
1. only **staff** of that org can access this api.
1. make swagger docs for the same 
1. add tests for the same

### PR DESCRIPTION
**Pull request by priyanshisharma**
### Description
I have created the API to display permission sets' id and name of the org to authenticated staff users along with tests and Swagger docs.

**Type of Change:** (Delete irrelevant options)
- Code
- Testing

**Code/Quality Assurance Only** 
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
I created relevant objects in my local, through Django Admin Panel, tested it with SureUtils REST API Client in all possible cases yielding all described outputs successfully. 

Organisation doesn't exist
Screen Shot 2020-05-24 at 1 40 49 AM

Organisation exists but user isn't a member of it
Screen Shot 2020-05-24 at 12 55 17 AM

Organisation exists and user is a member but since it doesn't have staff permissions it cannot view this.
Screen Shot 2020-05-24 at 12 55 43 AM

Successful submission
Screen Shot 2020-05-24 at 12 57 09 AM

I have also added tests, in `tests.org.test_list_permission_set_api.py` and receive the following on running `python3 manage.py test`.
```
Creating test database for alias 'default'...
System check identified no issues (0 silenced).
.....................................
----------------------------------------------------------------------
Ran 37 tests in 5.047s

OK
Destroying test database for alias 'default'...
```

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: org/views.py [hunk 1]
```diff
@@ -97,3 +97,50 @@ def AddVolunteer(request,org_id):
...
+    try:
+        member = Member.objects.get(
+        user = request.user,
+        org = org
+        )
+    except Member.DoesNotExist:
+        return Response({"detail" : "You are not a member of this organisation"}, status.HTTP_400_BAD_REQUEST)
+
+
+    if member.permissions.permissions.permissions[1]:
```
Comment (naveennvrgup) [inline org/views.py:129]:
```suggestion
    from org.custom_model_field import PermissionSet # on the top 
    if member.permissions.permissions.permissions[PermissionSet.IS_STAFF]: 
```

will be better way of doing this. 

We will rename **this ridiculous thing `permissions.permissions.permissions` in a separate issue lol**

Comment (naveennvrgup) [reply org/views.py:129]: Please create this as a separate **good first issue**.

**Review by naveennvrgup — APPROVED** [review]
1. Your code quality has improved remarkably.  :clap: 
1. Great testcases.
1. Tested locally & works great :fire:

File: org/views.py (hunk 1 above)
Comment (priyanshisharma) [reply org/views.py:129]: > will be better way of doing this.
>

I guess there is an ambiguity between the PermissionSet class and PermissionSet model due to which the tests fail when I make this change.. 😅 
`AttributeError: type object 'PermissionSet' has no attribute 'objects'` pops up when I refer to PermissionSet model below to get the respective ids and names.

Comment (priyanshisharma) [reply org/views.py:129]: > Please create this as a separate **good first issue**.
>
I think it is #92 what you're referring to..? 😅

Comment (priyanshisharma) [reply org/views.py:129]: > > will be better way of doing this.
> 
> I guess there is an ambiguity between the PermissionSet class and PermissionSet model due to which the tests fail when I make this change.. 😅
> `AttributeError: type object 'PermissionSet' has no attribute 'objects'` pops up when I refer to PermissionSet model below to get the respective ids and names.

Once this PR is merged, I'll change it accordingly when I change `custom_model_field.PermissionSet` to `custom_model_field.Permissions` while working on #92 , Does this sound right..?

**Comment by priyanshisharma** [comment]
> 1. Your code quality has improved remarkably.  👏
> 2. Great testcases.
> 3. Tested locally & works great 🔥

Thanks @naveennvrgup sir! 😄🎉

File: org/views.py (hunk 1 above)
Comment (naveennvrgup) [reply org/views.py:129]: > I guess there is an ambiguity between the PermissionSet class and PermissionSet model due to which the tests fail when I make this change.. sweat_smile
> AttributeError: type object 'PermissionSet' has no attribute 'objects' pops up when I refer to PermissionSet model below to get the respective ids and names.

Yup ambiguity. No need to create new issues. You can go on with #92. And u can resolve this conversation once u see it.

**Pull request merged**